#!/usr/bin/env python
PACKAGE = "cv"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#Color Masking parameters
gen.add("hueLowerBound",        int_t,    0, "Hue Lower Bound", 15,  0, 180)
gen.add("hueUpperBound",        int_t,    0, "Hue Lower Bound", 15,  0, 180)
gen.add("satLowerBound",        int_t,    0, "Hue Lower Bound", 15,  0, 255)
gen.add("satUpperBound",        int_t,    0, "Hue Lower Bound", 15,  0, 255)
gen.add("valLowerBound",        int_t,    0, "Hue Lower Bound", 15,  0, 255)
gen.add("valUpperBound",        int_t,    0, "Hue Lower Bound", 15,  0, 255)
gen.add("redChannelxfactor",    double_t, 0, 
        "Red channel increase factor"   , 1.5,1, 3)

exit(gen.generate(PACKAGE, "LaneDetector", "laneDetectorParams"))
